### DISCUSSION TITLE
More helpful error messages when Quarto chunk options are mis-specified

### DISCUSSION THREAD

**Discussion by AmeliaMN** [quarto-dev/quarto-cli#2433] [Feature Requests]
I posted this on the [RStudio community](https://community.rstudio.com/t/more-helpful-error-messages-when-quarto-options-are-mis-specified/147223) as well, but figured out it should actually be here. 

If you mess up when specifying your Quarto chunk options and use `=` instead of `:` in one of your code chunks (cough, former RMarkdown users), you will get a pretty impenetrable (to me) error about YAML. For example:

````
---
title: "Untitled"
format: pdf
editor: visual
---

```{r}
#| label: math
#| echo = false
2 * 2
```
````

Gives the error
```
processing file: test.qmd
Error in yaml::yaml.load(meta, handlers = list(expr = parse_only)) : 
  Scanner error: while scanning a simple key at line 2, column 1 could not find expected ':' at line 3, column 1
Calls: .main ... FUN -> parse_block -> partition_chunk -> <Anonymous>
Execution halted
```
I spent way too much time looking for errors in my header YAML, forgetting that those `#|` comments are basically YAML now, too. One of the problems is the `line 3` part. I was looking in the third line of my Quarto document!

Not sure if it would be possible, but it would be awesome to have an error message that said something like `Scanner error: while scanning the options in chunk 1, ...`

**Comment by cscheid** [comment]
This is indeed annoying. I wrote a custom handler exactly for this case a few months ago but for whatever reason it's not getting called. I'll create an issue to track.

**Reply by cderv** [reply]
Thanks for opening the issue https://github.com/quarto-dev/quarto-cli/issues/2434 @cscheid !

I'll lock the thread here, and we can all follow up in the issue directly. 

Thanks @AmeliaMN for the useful feedback !

**Reply by cscheid** [reply]
Fix incoming, new error message:

```
$ quarto render 2434.qmd
ERROR: Validation of YAML cell metadata failed.
(line 9, columns 4--16) property name echo = false is invalid

✖ The value echo = false is a string.
ℹ In YAML, key-value pairs in objects must be separated by a colon and a space.
ℹ Did you mean echo: false instead?

ERROR: Render failed due to invalid YAML.
```
